omod-1.9/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_9/VisitResource1_9.java

```diff
@@ -332,4 +337,12 @@ private Patient getPatient(String patientUniqueId) {
 			throw new ObjectNotFoundException();
 		return patient;
 	}
+	
+	private Location getLocation(String locationUniqueId) {
+		Location location = ((LocationResource1_8) Context.getService(RestService.class).getResourceByName(
+		    RestConstants.VERSION_1 + "/location")).getByUniqueId(locationUniqueId);
```

Comment:
Is there a reason not to just use the LocationService API directly here?  (Maybe there is a reason, I just haven't seen the pattern before)

Reply:
yeah, this is the first time i see this too. I just followed the pattern already used in this class to get the Patient parameter. Please read the entire java class.

Reply:
Cool, yeah, let's keep it then...